File: tokens/filter-tokens.py
import fileinput
MIN = 10

def replace(word):
	#TODO: see why this filter different from that in filter-actions.py
	singular = ["he", "she", "him", "i", "myself", "you", "me", "himself", "herself", "yourself"]
	plural = ["they", "them", "we", "us", "ourselves", "themselves"]
	if any(s == word for s in singular):
		return "the person"
	elif any(s == word for s in plural):
		return "the people"
	return word

object_match = ["the person", "the people", "it", "her",
	            "that", "this", "#", "there", "all", "'s", "the #"]

for line in fileinput.input():
	replace_pronouns = map(lambda x: ' '.join(map(lambda y: replace(y).rstrip() ,x.split(' '))), line.split('\t'))
	if not any(s == replace_pronouns[2].strip() for s in object_match) and int(replace_pronouns[3]) >= MIN:
		print '\t'.join(replace_pronouns)

File: legacy/pipeline/v0/filter-tokens.rb
# USE: ruby filter-tokens.rb all-tokens.tsv
# Throws away tokens thatinclude actions on other people, and tokens that don't occur frequently enough

MIN = 100

def replace(word)
  if ["he", "she", "him", "i", "myself", "you", "me", "himself", "herself", "yourself"].include?(word)
    "the person"
  elsif ["they", "them", "we", "us", "ourselves", "themselves"].include?(word)
    "the people"
  else
    word
  end
end

object_match = ["the person", "the people", "it", "her",
                "that", "this", "#", "there", "all", "'s", "the #"]

ARGF.each_line do |line|
  replace_pronouns = line.split("\t").map do |seg|
    seg.split(" ").map { |word| replace(word) }.join(" ")
  end
  if !object_match.include?(replace_pronouns[2].strip)
    puts replace_pronouns.join("\t") if replace_pronouns[3].to_i >= MIN
  end
end
